src/vs/workbench/contrib/mcp/browser/mcpResourceQuickAccess.ts

```diff
@@ -391,21 +395,45 @@ export abstract class AbstractMcpResourceAccessPick {
 		}));
 
 		store.add(picker.onDidAccept(async event => {
-			if (!event.inBackground) {
-				picker.hide(); // hide picker unless we accept in background
-			}
-
-			if (runOptions?.handleAccept) {
-				runOptions.handleAccept?.(picker.activeItems[0], event.inBackground);
-			} else {
-				const [item] = picker.selectedItems;
-				const uri = await helper.toURI((item as ResourceQuickPickItem).resource);
-				if (uri) {
-					this._editorService.openEditor({ resource: uri, options: { preserveFocus: event.inBackground } });
+			try {
+				picker.busy = true;
+				if (runOptions?.handleAccept) {
+					runOptions.handleAccept?.(picker.activeItems[0], event.inBackground);
+				} else {
+					const [item] = picker.selectedItems;
+					let uri: URI | undefined;
+					if (isDirectoryMode) {
+						uri = URI.parse((item as IQuickPickItem).id!);
+					} else {
+						uri = await helper.toURI((item as ResourceQuickPickItem).resource);
+					}
+					// If the URI is a directory, we need to read its contents and show them in the picker.
+					// directory URIs end with a `/`
+					if (uri?.path.endsWith('/')) {
+						try {
+							const stat = await this._fileService.resolve(uri, { resolveMetadata: false });
+							if (stat.children) {
+								isDirectoryMode = true;
+								const directoryItems: IQuickPickItem[] = stat.children.map((child) => ({
+									id: child.resource.toString(),
+									label: child.name,
+									description: child.isDirectory ? localize('mcp.directory', 'Directory') : '',
+								}));
+								picker.items = directoryItems;
+								picker.show();
+							}
+						} catch (error) {
+							this._notificationService.error(localize('mcp.readDirError', "Unable to read directory '{0}'", uri.toString()));
+						}
+					} else if (uri) {
+						picker.hide();
```

Comment:
We seem to have lost the `event.inBackground` option here. This can be tested by alt-clicking a resource in the Add Context flow.

Reply:
We do have event.inBackground present in this method.